opendata_collector_module/metrics_opendata_collector/opendata_api_client.py

```diff
@@ -0,0 +1,139 @@
+#  The MIT License
+#  Copyright (c) 2021- Nordic Institute for Interoperability Solutions (NIIS)
+#  Copyright (c) 2017-2020 Estonian Information System Authority (RIA)
+#
+#  Permission is hereby granted, free of charge, to any person obtaining a copy
+#  of this software and associated documentation files (the "Software"), to deal
+#  in the Software without restriction, including without limitation the rights
+#  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+#  copies of the Software, and to permit persons to whom the Software is
+#  furnished to do so, subject to the following conditions:
+#
+#  The above copyright notice and this permission notice shall be included in
+#  all copies or substantial portions of the Software.
+#
+#  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+#  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+#  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+#  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+#  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+#  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+#  THE SOFTWARE.
+
+import urllib
+from datetime import datetime
+
+import requests
+
+from metrics_opendata_collector.constants import (DT_FORMAT_W_TZ,
+                                                  DT_FORMAT_WO_TZ)
+
+DEFAULT_TZ_OFFSET = '+0000'
+
+
+class InputValidationError(Exception):
+    pass
+
+
+class OpenDataAPIClient:
+    """
+    A client for fetching data from an OpenData API.
+
+    Attributes:
+        _source_id (str): The source ID to fetch data from.
+        _settings_manager: The manager for getting OpenData API settings.
+        _source_settings (dict): The settings for the OpenData API source.
+        _url (str): The URL for the OpenData API.
+        timeout (int): The maximum time to wait for a response in seconds.
+    """
+
+    def __init__(self, source_id: str, settings_manager):
+        self._source_id = source_id
+        self._settings_manager = settings_manager
+        self._source_settings = (
+            self._settings_manager.get_opendata_source_settings(source_id)
+        )
+        self._url = self._source_settings['url']
+        self.timeout = self._settings_manager.settings.get('timeout') or 10
+
+    def get_query_params(self, overrides: dict = {}) -> dict:
```

Comment:
Default mutable parameters are strongly not recommended since they do mutate on different initializes, let's replace it with None value and inside the functions let's assign it to the empty dict then

Reply:
Fixed it